### PR TITLE
Continue processing other files if folder for given file is not configured

### PR DESCRIPTION
**Pull request by rkondratowicz**
just in case...

Removing the logic of determining the target folder from FTP client.
The caller sends the FTP client service folder that should be used.
When folder for given service is not found, this fact is logged and the client is never called.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java [hunk 1]
```diff
@@ -51,15 +49,13 @@ public FtpClient(
...
 
-    public void upload(LocalSourceFile file, boolean isSmokeTestFile, String service) {
+    public void upload(LocalSourceFile file, boolean isSmokeTestFile, String serviceFolder) {
         Instant now = Instant.now();
 
         runWith(sftp -> {
             boolean isSuccess = false;
 
             try {
-                String serviceFolder = getServiceFolderMapping(service);
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java]: 👍

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java [hunk 2]
```diff
@@ -79,20 +84,25 @@ public void run() {
...
+            );
+
+            ftp.upload(file, isSmokeTest(letter), serviceFolder.get());
+
+            logger.info(
+                "Uploaded letter id: {}, messageId: {}, file name: {}, additional data: {}",
+                letter.getId(),
+                letter.getMessageId(),
+                file.filename,
+                letter.getAdditionalData()
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java]: Is additional data not sensitive information?

Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java]: Those should claim reference or ccd case reference and it helps for debugging and analysing issues on prod. Services should not send any sensitive data afaik :)

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java [hunk 3]
```diff
@@ -79,20 +84,25 @@ public void run() {
...
-
-        logger.info(
-            "Uploaded letter id: {}, messageId: {}, file name: {}, additional data: {}",
-            letter.getId(),
-            letter.getMessageId(),
-            file.filename,
-            letter.getAdditionalData()
-        );
+        Optional<String> serviceFolder = serviceFolderMapping.getFolderFor(letter.getService());
+        if (serviceFolder.isPresent()) {
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java]: `ifPresent` could be used instead. Not much difference, up to you.

File: src/integrationTest/java/uk/gov/hmcts/reform/sendletter/services/FtpUploadTest.java [hunk 4]
```diff
@@ -26,28 +24,11 @@ public void uploads_file() throws Exception {
...
-            client.upload(doc, false, "bulkprint");
+            client.upload(doc, false, LocalSftpServer.SERVICE_FOLDER);
             File[] files = server.lettersFolder.listFiles();
             assertThat(files.length).isEqualTo(1);
             String content = new String(Files.toByteArray(files[0]));
             assertThat(content).isEqualTo("world");
         }
     }
-
-    @Test
```
Comment (lgonczar) [inline src/integrationTest/java/uk/gov/hmcts/reform/sendletter/services/FtpUploadTest.java]: We could also rename the name of this test, as it seems to be the client test. Just a suggestion.

Comment (rkondratowicz) [reply src/integrationTest/java/uk/gov/hmcts/reform/sendletter/services/FtpUploadTest.java]: you mean the class?

File: src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java (hunk 3 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java]: it can't, I need an `else`. `ifPresent` returns void.

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java]: true, what was I thinking?

File: src/integrationTest/java/uk/gov/hmcts/reform/sendletter/services/FtpUploadTest.java (hunk 4 above)
Comment (lgonczar) [reply src/integrationTest/java/uk/gov/hmcts/reform/sendletter/services/FtpUploadTest.java]: sorry, yes, test class

**Pull request merged**